backend/__tests__/posts.js

```diff
@@ -662,4 +663,88 @@ describe('Posts', () => {
             })
         })
     })
+
+    describe('GET /posts/:id/interactions', () => {
+        describe('when not authenticated', () => {
+            it('should return response code of 400', async () => {
+                const response = await request(server).get(
+                    '/api/posts/1/interactions'
+                )
+                expect(response.statusCode).toBe(400)
+            })
+        })
+
+        describe('when the post does not exist', () => {
+            it('should return response code of 404', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const response = await request(server)
+                    .delete(`/api/posts/99/interactions`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(404)
+            })
+        })
+
+        describe('when the post does exist', () => {
+            it('should return response code of 200', async () => {
+                const user1 = await models.users.create({
```

Comment:
You're repeating a lot of code here between tests - create some functions to create these users and authentication tokens.

Reply:
Yeah you're right, it is a lot. I've started creating some functions to help this however it is taking a while to rewrite all these tests. For now, I think it would be good to consider merging the request to be ready for the demo on Monday.